cdap-api/src/main/java/co/cask/cdap/api/schedule/NotificationSchedule.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+/**
+ * A schedule based on the reception of notifications.
+ */
+public interface NotificationSchedule extends Schedule {
+
+  /**
+   * @return Namespace of the emitting source of the notifications
+   */
+  String getNotificationSourceNamespaceId();
```

Comment:
We should in theory just have a ``Id.Feed getNotificationFeedId()`` method, but the ``Id`` class is in the ``proto`` module, not accessible from the ``cdap-api`` module. 
The notification feed is caracterized by 3 attributes, ``namespace``, ``category`` and ``name``. So I could have, instead of these two getters, ``String getNotificationFeedNamespaceId()``, ``String getNotificationFeedCategory()``, and ``String getNotificationFeedName()``.